File: online-judge-solutions/Codeforces/1481A.cpp
//https://codeforces.com/problemset/problem/1481/A
//Lang		:	GNU C++14
//Time		:	31 ms
//Memory	:	1300 KB
#include<iostream>
#include<algorithm>
using namespace std;
int main(){
	int t,px,py;
	cin>>t;
	while(t--){
		string s;
		cin>>px>>py>>s;
		int R=count(s.begin(),s.end(),'R');
		int L=count(s.begin(),s.end(),'L');
		int D=count(s.begin(),s.end(),'D');
		int U=count(s.begin(),s.end(),'U');
		bool rx=false,ry=false;
		if((px>=0 && R-px>=0) || (px<0 && px+L>=0)) rx=true;
		if((py>=0 && U-py>=0) || (py<0 && py+D>=0)) ry=true;
		(rx && ry) ? cout<<"YES\n":cout<<"NO\n";
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1481A.py
#https://codeforces.com/problemset/problem/1481/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	4700 KB
t=int(input())
for x in range(0,t):
    px,py=map(int,input().split())
    s=input()
    R=s.count('R')
    L=s.count('L')
    D=s.count('D')
    U=s.count('U')
    rx,ry=False,False
    if((px>=0 and R-px>=0) or (px<0 and px+L>=0)): rx=True
    if((py>=0 and U-py>=0) or (py<0 and py+D>=0)): ry=True
    print("YES") if (rx and ry) else print("NO")
